File: rb/features.rb
# McuCountdown SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module McuCountdownFeatures
  def self.make_feature(name)
    case name
    when "base"
      McuCountdownBaseFeature.new
    when "ratelimit"
      McuCountdownRatelimitFeature.new
    when "retry"
      McuCountdownRetryFeature.new
    when "test"
      McuCountdownTestFeature.new
    when "timeout"
      McuCountdownTimeoutFeature.new
    else
      McuCountdownBaseFeature.new
    end
  end
end

File: py/features.py
# McuCountdown SDK feature factory

from feature.base_feature import McuCountdownBaseFeature
from feature.test_feature import McuCountdownTestFeature


def _make_feature(name):
    features = {
        "base": lambda: McuCountdownBaseFeature(),
        "test": lambda: McuCountdownTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
